src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/PackageSourceMappingUtility.cs

```diff
@@ -0,0 +1,95 @@
+            IReadOnlyList<PackageSourceMappingSourceItem> existingPackageSourceMappingSourceItems)
+        {
+            if (userAction?.SourceMappingSourceName == null || addedPackages == null)
+            {
+                return;
+            }
+
+            Dictionary<string, IReadOnlyList<string>> patternsReadOnly = existingPackageSourceMappingSourceItems
+                .ToDictionary(pair => pair.Key, pair => (IReadOnlyList<string>)(pair.Patterns.Select(p => p.Pattern).ToList()));
+
+            PackageSourceMapping packageSourceMapping = new(patternsReadOnly);
+
+            // Expand all patterns/globs so we can later check if this package ID was already mapped.
+            IEnumerable<string> addedPackageIds = addedPackages.Select(action => action.Item1);
+            List<string> addedPackageIdsWithoutExistingMappings = new();
+            foreach (string? addedPackageId in addedPackageIds)
+            {
+                IReadOnlyList<string> configuredSource = packageSourceMapping.GetConfiguredPackageSources(addedPackageId);
+                if (configuredSource == null || configuredSource.Count == 0)
+                {
+                    addedPackageIdsWithoutExistingMappings.Add(addedPackageId);
+                }
+            }
+
+            // Get all newly added package IDs that were not previously Source Mapped.
+            // Always include the Package ID being installed since it takes precedence over any globbing.
+            string[] packageIdsNeedingNewSourceMappings = addedPackageIdsWithoutExistingMappings
+                .Union(new string[] { userAction.PackageId })
+                .ToArray();
+
+            CreateAndSavePackageSourceMappings(
+                sourceName: userAction.SourceMappingSourceName,
+                newPackageIdsToSourceMap: packageIdsNeedingNewSourceMappings,
+                sourceMappingProvider,
+                existingPackageSourceMappingSourceItems);
+        }
+
+        private static void CreateAndSavePackageSourceMappings(
+            string sourceName,
+            string[] newPackageIdsToSourceMap,
+            PackageSourceMappingProvider mappingProvider,
+            IReadOnlyList<PackageSourceMappingSourceItem> existingPackageSourceMappingSourceItems)
+        {
+            if (string.IsNullOrWhiteSpace(sourceName) || newPackageIdsToSourceMap is null || newPackageIdsToSourceMap.Length == 0)
+            {
+                return;
+            }
+
+            IEnumerable<PackagePatternItem> newPackagePatternItems = newPackageIdsToSourceMap.Select(packageId => new PackagePatternItem(packageId));
+
+            List<PackageSourceMappingSourceItem> newAndExistingPackageSourceMappingItems = new(existingPackageSourceMappingSourceItems);
+
+            PackageSourceMappingSourceItem packageSourceMappingItemForSource =
+                        existingPackageSourceMappingSourceItems
+                        .Where(mappingItem => mappingItem.Key == sourceName)
```

Comment:
This is a great question. My assumption was that it should be explicitly the casing of the source requested.
I double-checked and fascinatingly, you **can** provide different casing of the same source name when hand-editing a nuget.config. I got no errors in VS from loading a config with nuget.org and NuGet.OrG both existing in the same config but point to different servers.

Using VS Options does show an error if you attempt to add a different casing of an existing source through the Options UI.

Therefore, since we don't have guarantees that there are unique names, I think this utility should enforce case checking as written..